azure-storage-blob/azure/storage/blob/blockblobservice.py

```diff
@@ -574,6 +574,20 @@ def create_blob_from_stream(
                 progress_callback(0, count)
 
             data = stream.read(count)
+            data_chunk = data  # to store the chunk of data read from stream each time
+
+            # keep reading from stream util length of data >= count or reaching the end of stream
+            while len(data) < count and len(data_chunk) is not 0:
+                data_chunk = stream.read(count)
+                data += data_chunk
+
+            if len(data) < count:
+                raise ValueError('Parameter:count is greater than the amount of data in the stream,'
+                                 'please specify a valid count')
+
+            if len(data) > count:
```

Comment:
Ideally we should adjust the count when reading (`stream.read(count)`) so that we never read more than necessary. In this case, we already read more than necessary and are chopping it off.

Reply:
I figured it was fine since the right count is eventually used because of the truncation, but you're probably right that reading extra might consume data from the stream that the customer wanted to send somewhere else.

Reply:
then do we still want to upload data in this case? **_stream.read(count)_** is never reading the right count, that's why we have a while loop to keep reading.

Reply:
We discussed offline. Basically, if x bytes are desired, but only y bytes are read, then we should read x-y bytes next time, and keep trying to read the remainder until we have x bytes in total.